server/plugin.go

```diff
@@ -1,28 +1,117 @@
 package main
 
 import (
+	"encoding/json"
 	"fmt"
-	"net/http"
+	"io/ioutil"
+	"path/filepath"
 	"sync"
 
+	"github.com/mattermost/mattermost-plugin-api/experimental/command"
+	"github.com/mattermost/mattermost-server/v5/model"
 	"github.com/mattermost/mattermost-server/v5/plugin"
+	"github.com/ulumuri/go-nagios/nagios"
 )
 
-// Plugin implements the interface expected by the Mattermost server to communicate between the server and plugin processes.
+// Plugin implements the interface expected by the Mattermost server to
+// communicate between the server and plugin processes.
 type Plugin struct {
 	plugin.MattermostPlugin
 
 	// configurationLock synchronizes access to the configuration.
 	configurationLock sync.RWMutex
 
-	// configuration is the active plugin configuration. Consult getConfiguration and
-	// setConfiguration for usage.
+	// configuration is the active plugin configuration. Consult
+	// getConfiguration and setConfiguration for usage.
 	configuration *configuration
+
+	client *nagios.Client
+
+	botUserID string
+
+	commandHandlers map[string]commandHandlerFunc
+}
+
+func (p *Plugin) setDefaultKV(key string, value interface{}) error {
+	b, err := json.Marshal(value)
+	if err != nil {
+		return fmt.Errorf("json.Marshal: %w", err)
+	}
+
+	if err := p.API.KVSet(key, b); err != nil {
+		return fmt.Errorf("p.API.KVSet: %w", err)
+	}
+
+	return nil
+}
+
+var defaultKVStore = map[string]interface{}{
```

Comment:
why not have these in the plugin settings?

Reply:
Done.